config/platform.go

```diff
@@ -0,0 +1,106 @@
+package config
+
+import (
+	"bufio"
+	"context"
+	"os"
+	"os/exec"
+	"regexp"
+	"runtime"
+	"strings"
+	"time"
+
+	"go.viam.com/rdk/logging"
+)
+
+var (
+	cudaRegex         = regexp.MustCompile(`Cuda compilation tools, release (\d+)\.`)
+	dpkgVersionRegex  = regexp.MustCompile(`\nVersion: (\d+)\D`)
+	savedPlatformTags []string
+)
+
+// helper to read platform tags for GPU-related system libraries.
+func readGPUTags(tags []string) []string {
+	// this timeout is for all steps in this function.
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+	if _, err := exec.LookPath("nvcc"); err == nil {
+		out, err := exec.CommandContext(ctx, "nvcc", "--version").Output()
+		if err != nil {
+			logging.Global().Errorw("error getting cuda version from nvcc. cuda-specific modules may not load", "err", err)
```

Comment:
Discussion question: should it be capitalized as cuda or CUDA? I can argue either way.

Reply:
I updated everywhere to `Cuda` based on what `nvcc` outputs